src/main/java/com/blackducksoftware/integration/hub/alert/web/actions/ConfigActions.java

```diff
@@ -63,26 +64,68 @@ public boolean doesConfigExist(final Long id) {
             if (foundEntity != null) {
                 final R restModel = objectTransformer.databaseEntityToConfigRestModel(foundEntity, configRestModelClass);
                 if (restModel != null) {
-                    return Arrays.asList(restModel);
+                    final R maskedRestModel = maskRestModel(restModel);
+                    return Arrays.asList(maskedRestModel);
                 }
             }
             return Collections.emptyList();
         }
         final List<D> databaseEntities = repository.findAll();
         final List<R> restModels = objectTransformer.databaseEntitiesToConfigRestModels(databaseEntities, configRestModelClass);
-        return restModels;
+        return maskRestModels(restModels);
+    }
+
+    public R maskRestModel(final R restModel) {
+        return restModel;
+    }
+
+    public List<R> maskRestModels(final List<R> restModels) {
+        final List<R> maskedRestModels = new ArrayList<>();
+        for (final R restModel : restModels) {
+            maskedRestModels.add(maskRestModel(restModel));
+        }
+        return maskedRestModels;
     }
 
     public void deleteConfig(final String id) {
         deleteConfig(objectTransformer.stringToLong(id));
     }
 
     public void deleteConfig(final Long id) {
-        if (id != null) {
+        if (null != id) {
             repository.delete(id);
         }
     }
 
+    public D updateNewConfigWithSavedConfig(final D newConfig, final String id) {
+        if (StringUtils.isNotBlank(id)) {
+            final Long longId = objectTransformer.stringToLong(id);
+            final D savedConfig = repository.findOne(longId);
+            return updateNewConfigWithSavedConfig(newConfig, savedConfig);
+        }
+        return newConfig;
+    }
+
+    public D updateNewConfigWithSavedConfig(final D newConfig, final D savedConfig) {
```

Comment:
This gets overridden if needed by the actions to use saved fields if the newConfig has empty "hidden" fields like passwords. We could make this abstract and force all actions to implement, ideas?

Reply:
After looking at this class with more context, I think our best bet is to abstract and force the implementation. In general, I don't like implementing something that will always be overridden. The same seems to apply to for the comment above.